doc/How-Tos/DependencyInjection/DependencyInjection-setup.md

```diff
@@ -0,0 +1,88 @@
+# How-To: Use Services with Dependency Injection
+
+Dependency injection (DI) is an important design pattern for building loosely-coupled software that allows for maintainability and testing. This tutorial will walk you through how to register services with a service provider for use throughout your application.
+
+NOTE: This guide assumes you used the Uno.Extensions `dotnet new` template to create the solution. You can verify this by looking for a generated `App.xaml.host.cs` file.
+
+## Step-by-steps
+
+1. Plan the contract for your service
+    * Create a new interface which declares the method(s) your service offers:
+    ```cs
+        public interface IUserProfilePictureService
+        {
+	        Task<byte[]> GetAsync(CancellationToken ct = null);
```

Comment:
Should we already imply nullables to be enabled?
```suggestion
	        Task<byte[]> GetAsync(CancellationToken? ct = null);
```

Reply:
Actually I think it would be better practice to remove the default completely here
Task<byte[]> GetAsync(CancellationToken ct);

Also might be worth simplifying the name of this service to just IProfilePictureService